src/libs/sfdk/virtualmachine_p.h

```diff
@@ -189,6 +186,9 @@ class VirtualMachinePrivate
 private:
     void enableUpdates();
 
+signals:
+    void aboutToRestoreSnapshot(const std::shared_ptr<bool> &ok);
```

Comment:
Somehow it feels a bit wrong to have a signal which, based on the given name, seems to just indicate that we are about to restore a snapshot, to have a return value as a parameter. I mean, it's not like we are going to not restore the snapshot because the slot failed to do whatever it wants to do? I do understand that it's there because we want to forward the error reporting, but still it feels a bit weird.

Reply:
Yes, the intention here is not just signaling but calling back and getting the return value(s) collected from all observers. Moreover an asynchronous way - hence the `shared_ptr`. (It would be plain `bool *ok` in synchronous code). Compare this e.g. with Boost's combiners https://www.boost.org/doc/libs/1_61_0/doc/html/signals2/tutorial.html#idp394578384 . Qt is not that powerful here, so the combiner work is offloaded to observers.

Reply:
Interesting read, thanks for the link. The signals in boost are indeed a bit different than the signals in Qt. You are now twisting the Qt signals to act more like boost signals. I'm not saying it's wrong, it just feels a bit weird. And now I'm also a bit sad the Qt didn't take the opportunity to add signal return values (and combiners!) in Qt 6. Because those would be useful, like we see here!